runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java

```diff
@@ -433,4 +455,512 @@ public void onExecution(Instruction instruction, long[] operands, MStack stack)
             listener.onExecution(instruction, operands, stack);
         }
     }
+
+    public static Builder builder(Module module) {
+        return new Builder(module);
+    }
+
+    public static final class Builder {
+        private final Module module;
+
+        private boolean initialize = true;
+        private boolean start = true;
```

Comment:
All you said is correct, this flag means:
if the module has a start function and this flag is set to "false" it will not be run on instantiation.
I think that we found it useful in some situations to be able to tweak this aspect.